eureka-client/src/test/java/com/netflix/discovery/guice/EurekaModuleTest.java

```diff
@@ -0,0 +1,86 @@
+package com.netflix.discovery.guice;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.Scopes;
+import com.google.inject.util.Modules;
+import com.netflix.appinfo.ApplicationInfoManager;
+import com.netflix.appinfo.EurekaInstanceConfig;
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.appinfo.providers.MyDataCenterInstanceConfigProvider;
+import com.netflix.config.ConfigurationManager;
+import com.netflix.discovery.DiscoveryClient;
+import com.netflix.discovery.DiscoveryManager;
+import com.netflix.discovery.EurekaClient;
+import com.netflix.discovery.EurekaClientConfig;
+import com.netflix.governator.guice.LifecycleInjector;
+import com.netflix.governator.guice.LifecycleInjectorBuilder;
+import com.netflix.governator.lifecycle.LifecycleManager;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author David Liu
+ */
+public class EurekaModuleTest {
+
+    private Injector injector;
+    private LifecycleManager lifecycleManager;
+
+    @Before
+    public void setUp() throws Exception {
+        ConfigurationManager.getConfigInstance().setProperty("eureka.region", "default");
+        ConfigurationManager.getConfigInstance().setProperty("eureka.shouldFetchRegistry", "false");
+        ConfigurationManager.getConfigInstance().setProperty("eureka.registration.enabled", "false");
+        ConfigurationManager.getConfigInstance().setProperty("eureka.serviceUrl.default", "http://localhost:8080/eureka/v2");
```

Comment:
We'd have to mock a few different configs. This also allows us to test the actual config bindings. The test runs pretty fast even with the archaius singleton.